codex/sales/salesagent.nim

```diff
@@ -143,6 +143,6 @@ proc unsubscribe*(agent: SalesAgent) {.async.} =
 
   agent.subscribed = false
 
-proc stop*(agent: SalesAgent) {.async.} =
+proc stop*(agent: SalesAgent) {.async: (raises: [CancelledError]).} =
```

Comment:
This routine as well should not be cancellable imo, since it invokes teardown mechanisms.
```suggestion
proc stop*(agent: SalesAgent) {.async: (raises: []).} =
```